analytics/js_tracking_api.rst

```diff
@@ -0,0 +1,859 @@
+
+    **Optional**  :ref:`Custom dimension<Custom Dimensions>` that should be tracked with this action. Can be multiple dimensions.
+    Written as object property using ``dimension{ID}`` notation.
+
+    Example::
+
+        {
+           dimension1: "example value",
+           dimension2: "example value"
+        }
+
+
+Example of usage
+
+.. code-block:: html
+
+    <button onClick = function(){_paq.push(["trackLink", "http://www.example.com/example", "link"]);}>
+        Click me!
+    </button>
+
+Tracking Downloads
+``````````````````
+
+Following extensions are tracked as download by default:
+
+
++-------+-----+-----+-----+------+-----+-----+-----+------+-----+------+-----+---------+-----+-----+
+| 7z    | aac | arc | arj | apk  | asf | asx | avi | bin  | bz  | bz2  | csv | deb     | dmg | doc |
++-------+-----+-----+-----+------+-----+-----+-----+------+-----+------+-----+---------+-----+-----+
+| exe   | flv | gif | gz  | gzip | hqx | jar | jpg | jpeg | js  | mp2  | mp3 | mp4     | mpg | mov |
++-------+-----+-----+-----+------+-----+-----+-----+------+-----+------+-----+---------+-----+-----+
+| movie | msi | msp | odb | odf  | odg | odp | ods | odt  | ogg | ogv  | pdf | phps    | png | ppt |
++-------+-----+-----+-----+------+-----+-----+-----+------+-----+------+-----+---------+-----+-----+
+| qt    | qtm | ra  | ram | rar  | rpm | sea | sit | tar  | tbz | tbz2 | tgz | torrent | txt | wav |
++-------+-----+-----+-----+------+-----+-----+-----+------+-----+------+-----+---------+-----+-----+
+| wma   | wmv | wpd | xls | xml  | z   | zip |     |      |     |      |     |         |     |     |
++-------+-----+-----+-----+------+-----+-----+-----+------+-----+------+-----+---------+-----+-----+
+
+You can add extension to this list using ``addDownloadExtensions`` function::
+
+    _paq.push(["addDownloadExtensions", extensions]);
+
+.. data:: extensions
+
+    **Required** String containing extensions separated by ``|`` for example ``"mhj|docx"``
+
+
+Example of usage::
+
+    _paq.push(["addDownloadExtensions", "mhj|docx"]);
+
+This list can be rewrote using ``setDownloadExtensions`` function::
+
+    _paq.push(["setDownloadExtensions", extensions]);
+
+.. data:: extensions
+
+    **Required** String containing extensions separated by ``|`` for example ``"7z|apk|mp4"``
+
+
+Example of usage::
+
+    _paq.push(["setDownloadExtensions", "7z|apk|mp4"]);
+
+To track clicking a link as an download using css class simply add ``piwik_download`` class to link element.
+
+This class name can be changed, use ``setDownloadClasses`` to define which CSS class should be tracked::
+
+    _paq.push(["setDownloadClasses", className]);
+
+
+.. data:: className
+
+    **Required** String with css class that should be tracked instead of ``piwik_download``
+
+Example of usage::
+
+    _paq(["setDownloadClasses", "track-this-link-for-download"]);
+
+
+If you want to use JS instead you can add ``trackLink`` function to element ``onClick`` attribute::
```

Comment:
This function is declared a second time. Maybe it would be better to define it in one place and link to it as alternative to automatic link/download tracking.